### PR TITLE
[MRESOLVER-358] Upgrade some dependencies

### PR DESCRIPTION
**Pull request by cstamas**
Upgrades
* guava to 31.1-jre (was 30.1-jre)
* mockito-core to 4.11.0 (was 4.8.1)
* plexus-utils to 3.5.1 (was 3.5.0)
* maven deps (used in demos only) to 3.8.7 (was 3.8.4)

---

https://issues.apache.org/jira/browse/MRESOLVER-358

### CODE REVIEW

File: maven-resolver-demos/pom.xml [hunk 1]
```diff
@@ -38,7 +38,7 @@
   </modules>
 
   <properties>
-    <mavenVersion>3.8.4</mavenVersion>
+    <mavenVersion>3.8.8</mavenVersion>
```
Comment (slawekjaranowski) [inline maven-resolver-demos/pom.xml:41]: There is IT in resolver-demo-maven-plugin which will fail when Maven < 3.8.8 will be used ...
But we enforce minimal version as 3.8.4

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

File: maven-resolver-demos/pom.xml (hunk 1 above)
Comment (cstamas) [reply maven-resolver-demos/pom.xml:41]: Unsure I get it: ITs ar built by CI and also by me locally. Which IT will fail, as there is none failing for me?

Comment (cstamas) [reply maven-resolver-demos/pom.xml:41]: Locally I have no IT failure either, so what is the problem exactly?

**Review by slawekjaranowski — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #1034

**Comment by jira-importer** [comment]
Resolve #1034

**Pull request merged**